plugins/auth-backend/migrations/20220522100910_key_field_size.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function up(knex) {
+  // Sqlite does not support alter column.
+  if (!knex.client.config.client.includes('sqlite3')) {
```

Comment:
Do we to implement an option for SQLite as well? as in does SQLite not actually limit the size to begin with?

Reply:
I think Knex will set this field to VARCHAR(255) according to [this](https://github.com/knex/knex/issues/4409#:~:text=Currently%2C%20Knex%20gives%20a%20default,string()%20.).
If we need to support this migration also for SQLite then in the `up` function we need to remove the whole `signing_keys` table and create a new one from scratch (due to lack of alter column). 
Do you see any problem in doing this?

Reply:
If it helps: Sqlite3 ignores the length in the varchar columns. It's there only for compatibility.

Reply:
From the [sqlite3 specs](https://www.sqlite.org/datatype3.html):

**"Note that numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions (other than the large global [SQLITE_MAX_LENGTH](https://www.sqlite.org/limits.html#max_length) limit) on the length of strings, BLOBs or numeric values."**

So - even if Knex created a `varchar(255)` for Sqlite3 - the column is the equivalent of `varchar(SQLITE_MAX_LENGTH)`.  That's why I hit the bug only when deploying 😮

Reply:
@manuelscurti BTW: awesome job!